client/verifier_test.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package client
+
+import (
+	"testing"
+
+	"github.com/google/trillian"
+	tcrypto "github.com/google/trillian/crypto"
+	"github.com/google/trillian/crypto/keys"
+	"github.com/google/trillian/merkle/rfc6962"
+	"github.com/google/trillian/testonly"
+	"strings"
+)
+
+func TestVerifyRootErrors(t *testing.T) {
+	// Test setup
+	key, err := keys.NewFromPrivatePEM(testonly.DemoPrivateKey, testonly.DemoPrivateKeyPass)
+	if err != nil {
+		t.Fatalf("Failed to open test key, err=%v", err)
+	}
+	signer := tcrypto.NewSHA256Signer(key)
+	pk, err := keys.NewFromPublicPEM(testonly.DemoPublicKey)
+	if err != nil {
+		t.Fatalf("Failed to load public key, err=%v", err)
+	}
+
+	signedRoot := trillian.SignedLogRoot{0, nil, 0, nil, 0, 0}
```

Comment:
this could be rewritten as `trillian.SignedLogRoot{}` since all the fields have their default values